src/daemons/linuxd/src/main.rs

```diff
@@ -102,87 +97,85 @@ pub fn main() -> Result<()> {
         None => DEFAULT_BIND_SOCKET_TYPE,
     };
 
+    let gateway_bind_socket_type: SocketType = match args.gateway_bind_socket_type() {
+        Some(typ) => match SocketType::from_str(typ.as_str()) {
+            Ok(typ) => typ,
+            Err(error) => {
+                error!("{error} (type={typ:?})");
+                anyhow::bail!("failed to parse socket address type");
+            },
+        },
+        None => DEFAULT_GATEWAY_SOCKET_TYPE,
+    };
+
     let uvm_listener: SocketListener = match Socket::bind(uvm_bind_socket_type, uvm_sockaddr.clone()) {
         Ok(listener) => listener,
         Err(e) => {
-            error!("failed to bind to uVM socket address (error={e:?})");
+            error!("failed to bind to uVM socket address (address={}, error={e:?})", uvm_sockaddr.clone());
             anyhow::bail!("failed to bind to uVM socket address");
         },
     };
 
-    // Install signal handler.
```

Comment:
We do not need the signal handler anymore, because the `Drop` trait introduced in the previous commit will take care of the clean-up. Also note that this handler was an incomplete clean-up, as it was not covering the cases where an internal error happened after binding to a socket.